ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ObjectDiffusion/Outbound.hs

```diff
@@ -100,17 +107,19 @@ objectDiffusionOutbound tracer maxFifoLength ObjectPoolReader{..} _version =
 
   updateStNewObjects ::
     OutboundSt objectId object ticketNo ->
-    [(object, ticketNo)] ->
+    Map ticketNo object ->
     OutboundSt objectId object ticketNo
   updateStNewObjects !OutboundSt{..} newObjectsWithTicketNos =
     -- These objects should all be fresh
-    assert (all (\(_, ticketNo) -> ticketNo > lastTicketNo) newObjectsWithTicketNos) $
+    assert (all (\ticketNo -> ticketNo > lastTicketNo) (Map.keys newObjectsWithTicketNos)) $
```

Comment:
NIT: Now that the tuple is gone, I think  `all (> lastTicketNo)` would roll off the tongue a bit better.